includes/Block.php

```diff
@@ -1225,4 +1227,45 @@ public function setBlockEmail( $blockEmail ) {
 	public function setCreateAccount( $createAccount ) {
 		$this->mCreateAccount = $createAccount;
 	}
+
+	/**
+	 * Wikia change
+	 * SUS-288: Hide blocker name from logs and error pages if the block was made by staff/VSTF
+	 * @return bool Whether to hide the blocker's user name
+	 */
+	public function shouldHideBlockerName() {
+		$blocker = ( $this->blocker instanceof User ) ? $this->blocker : User::newFromName( $this->blocker );
+
+		if ( $blocker instanceof User ) {
+			$this->blocker = $blocker;
+			return $this->permissionsService()->hasPermission( $blocker, 'hideblockername' );
+		}
+
+		return false;
+	}
+
+	/**
+	 * SUS-288: Return the group name that should be shown instead of user name if the blocker name is hidden
+	 * (i.e. if the block was made by staff/VSTF)
+	 * @return string Group name text that will be parsed and output on the error page
+	 */
+	public function getGroupNameForHiddenBlocker() {
+		/** @var User $blockerUser */
+		$blockerUser = $this->getBlocker();
+		$permissionsService = $this->permissionsService();
+
+		// Get the global groups of this user that have 'hideblockername' permission
+		$groups = array_intersect(
+			$permissionsService->getExplicitGlobalGroups( $blockerUser ),
+			$permissionsService->getConfiguration()->getGroupsWithPermission( 'hideblockername' )
+		);
+
+		// Select the group name to show in the block message
+		if ( count( $groups ) ) {
+			$group = array_shift( $groups );
+			return wfMessage( "group-$group" )->plain();
```

Comment:
I don't like referring to message keys by string concatenation as it makes it very hard to find the usage in the code by message key. Imagine searching for all usages of `group-admin` - it will skip this place. We need to figure out something different. Note, that the set of names is limited.